services/src/main/java/org/keycloak/authentication/authenticators/client/JWTClientValidator.java

```diff
@@ -116,6 +116,13 @@ public void readJws() throws JWSInputException {
 
         jws = new JWSInput(clientAssertion);
         token = jws.readJsonContent(JsonWebToken.class);
+
+        var event = context.getEvent();
+        if (token != null && event != null) {
+            event.detail("client_assertion_jti", token.getId());
+            event.detail("client_assertion_issuer", token.getIssuer());
+            event.detail("client_assertion_subject", token.getSubject());
```

Comment:
We have constants like `Details.TOKEN_ID` to place a token ID into an event. Can it be used here as well?